integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/CacheControlHeadersTestCase.java

```diff
@@ -0,0 +1,149 @@
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import javax.xml.namespace.QName;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This class is to verify that cache mediator honor cache-control headers when enableCacheControl, includeAgeHeader
+ * is set to true in it's configuration
+ */
+public class CacheControlHeadersTestCase extends ESBIntegrationTest {
+
+    private String messagePayload;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+
+        super.init();
+        loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/cache/CacheControlHeaders.xml");
+        messagePayload = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" " +
+                "xmlns:ser=\"http://services.samples\" xmlns:xsd=\"http://services.samples/xsd\">\n" +
+                "   <soapenv:Header/>\n" +
+                "   <soapenv:Body>\n" +
+                "      <ser:getQuote>\n" +
+                "         <ser:request>\n" +
+                "            <xsd:symbol>WSO2</xsd:symbol>\n" +
+                "         </ser:request>\n" +
+                "      </ser:getQuote>\n" +
+                "   </soapenv:Body>\n" +
+                "</soapenv:Envelope>";
+    }
+
+    /**
+     * This test case checks whether the  back-end (axis2Server) has been hit for each request as no-store header is
+     * returned with the response.
+     */
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    @Test(groups = {"wso2.esb"}, description = "Testing with a large cache time out value for cache mediator")
+    public void testNoStore() throws Exception {
+        String apiName = "noStore";
+        OMElement response;
+        response = axis2Client.sendSimpleStockQuoteRequest(null, getApiInvocationURL(apiName), "");
+
+        String changeValue = response.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        response = axis2Client.sendSimpleStockQuoteRequest(null, getApiInvocationURL(apiName), "");
+
+        String newChangeValue = response.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        //The 'change' value should not be equal to the initial 'change' value as the response is not cached.
+        assertNotEquals(changeValue, newChangeValue, "Cache mediator cached the response");
```

Comment:
Here, what we are checking is whether the response is cached or not. Since no-store header exists, the response should not be cached. If it is cached we will get the same value for both requests. Do you think still we need to print the newChangeValue?

Reply:
+1 No need to print the value in that case